vertx-auth-oauth2/src/main/java/io/vertx/ext/auth/oauth2/providers/Auth0Auth.java

```diff
@@ -0,0 +1,67 @@
+package io.vertx.ext.auth.oauth2.providers;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.JWTOptions;
+import io.vertx.ext.auth.oauth2.OAuth2Auth;
+import io.vertx.ext.auth.oauth2.OAuth2Options;
+
+/**
+ * Simplified factory to create an {@link OAuth2Auth} for Okta Auth0.
+ *
+ * @author <a href="mailto:alexei.klenin@gmail.com">Alexei KLENIN</a>
+ */
+public interface Auth0Auth extends OpenIDConnectAuth {
+
+  /**
+   * Create a OAuth2Auth provider for Okta Auth0
+   *
+   * @param clientId     the client id given to you by Okta Auth0
+   * @param clientSecret the client secret given to you by Okta Auth0
+   * @param tenant       the tenant
+   * @param audience     the audience
+   */
+  static OAuth2Auth create(Vertx vertx, String clientId, String clientSecret, String tenant, String audience) {
+    return create(vertx, clientId, clientSecret, tenant, audience, new HttpClientOptions());
+  }
+
+  /**
+   * Create a OAuth2Auth provider for Okta Auth0
+   *
+   * @param clientId          the client id given to you by Okta Auth0
+   * @param clientSecret      the client secret given to you by Okta Auth0
+   * @param tenant            the tenant
+   * @param audience          the audience
+   * @param httpClientOptions custom http client options
+   */
+  static OAuth2Auth create(
+      Vertx vertx,
+      String clientId,
+      String clientSecret,
+      String tenant,
+      String audience,
+      HttpClientOptions httpClientOptions) {
+    if (audience == null) {
+      throw new IllegalStateException("audience cannot be null");
+    }
```

Comment:
```suggestion
    Objects.requireNonNull(audience, "audience cannot be null");
```

ISE is not the exception type we use for that kind of checks.

Reply:
Great observation! It should rather be `IllegalArgumentException`, but I messed up with IDE autocompletion. Will fix in no time.